File: Solution/1431.拥有最多糖果的孩子/1431.拥有最多糖果的孩子.cpp
/*
 * @Descroption: LeetCode 1431. 拥有最多糖果的孩子
 * @Author: EmoryHuang
 * @Date: 2021-10-20 19:35:50
 * @Method:
 * 简单题，遍历数组与最大值比较
 */

class Solution {
   public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int maxc = *max_element(candies.begin(), candies.end());
        for (int num : candies) ans.push_back(num + extraCandies >= maxc);
        return ans;
    }
};

File: Solution/1431.拥有最多糖果的孩子/1431.拥有最多糖果的孩子.py
'''
Descroption: LeetCode 1431. 拥有最多糖果的孩子
Author: EmoryHuang
Date: 2021-10-20 19:31:33
Method:
简单题，遍历数组与最大值比较
'''


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxc = max(candies)
        return [num + extraCandies >= maxc for num in candies]